src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -285,10 +285,22 @@ class Ledger(vm: VM) extends Logger {
     * @param worldStateProxy
     * @return
     */
-  private[ledger] def updateSenderAccountBeforeExecution(stx: SignedTransaction, worldStateProxy: InMemoryWorldStateProxy): InMemoryWorldStateProxy = {
+  private[ledger] def updateSenderAccountBeforeExecution(stx: SignedTransaction, worldStateProxy: InMemoryWorldStateProxy):
+  (InMemoryWorldStateProxy, Address) = {
+    val (world, recipient) = stx.tx.receivingAddress match {
+      case None =>
+        val (address, world1) = worldStateProxy.newAddress(stx.senderAddress)
+        val world2 = world1.newEmptyAccount(address)
+        val world3 = world2.transfer(stx.senderAddress, address, UInt256(stx.tx.value))
+        (world3, address)
+      case Some(txReceivingAddress) =>
+        val world1 = worldStateProxy.transfer(stx.senderAddress, txReceivingAddress, UInt256(stx.tx.value))
+        (world1, txReceivingAddress)
+    }
```

Comment:
You cannot do this here. This is the state that is reverted to in case of a VM exception. Value transfer should not be included in that state.